src/main/java/seedu/address/logic/parser/TimeParser.java

```diff
@@ -37,22 +40,98 @@ public class TimeParser {
         }
 
-        if (matchToday.matches()) {
-            LocalDate today = LocalDate.now();
-            String todayString = today.format(DATE_FORMAT);
-            return new AppointmentTime(todayString + " " + args.split(" ")[1].trim());
+        if (matchToday.matches()) { // format of command is d/tdy TIME
+            if (uppercaseAppointmentTime.startsWith("TODAY")) { // Format: today time
+                if (validAppointmentWindow(uppercaseAppointmentTime.substring(6))) {
+                    LocalDate today = LocalDate.now();
+                    String todayString = today.format(DATE_FORMAT);
+                    return new AppointmentTime(todayString + " " + args.split(" ")[1].trim());
+                } else {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TimeParser.MESSAGE_USAGE));
+                }
+            } else if (uppercaseAppointmentTime.startsWith("TDY")) {
+                if (validAppointmentWindow(uppercaseAppointmentTime.substring(4))) {
+                    LocalDate today = LocalDate.now();
+                    String todayString = today.format(DATE_FORMAT);
+                    return new AppointmentTime(todayString + " " + args.split(" ")[1].trim());
+                } else {
+                    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, TimeParser.MESSAGE_USAGE));
+                }
+            }
         }
 
         return new AppointmentTime(args);
     }
+
+    /**
+     * Checks a time to make sure it is valid.
+     *
+     * @return boolean
+     */
+    public static boolean validAppointmentWindow(String args) {
+        String[] times = args.replace(" ", "").toUpperCase().split("-");
+        assert times.length == 2;
+
+        int startHour = Integer.parseInt(times[0].substring(0, times[0].length() - 2));
+        assert startHour > 0;
+        assert startHour <= 12;
+
+        if (times[0].equals("12AM")) {
+            startHour = 0;
+        } else if (times[0].equals("12PM")) {
+            startHour = 12;
+        } else if (times[0].endsWith("PM")) {
+            startHour += 12;
+        }
+
+        int endHour = Integer.parseInt(times[1].substring(0, times[1].length() - 2));
+        assert endHour > 0;
+        assert endHour <= 12;
+
+        if (times[1].equals("12AM")) {
+            endHour = 0;
+        } else if (times[1].equals("12PM")) {
+            endHour = 12;
+        } else if (times[1].endsWith("PM")) {
+            endHour += 12;
+        }
+
+        return endHour > startHour;
+    }
+
+    /**
+     * Fill in later.
+     *
+     * @return boolean
+     */
+    public static boolean validAppointmentDate(String args) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+        sdf.setLenient(false);
+        try {
+            Date date = sdf.parse(args);
+            System.out.println(date.after(sdf.parse("06/02/1819")));
```

Comment:
can remove the log lines after u are done!

Reply:
Thanks for the spot! For the other two above, have changed the internal logic - the regex now only needs to accept something of the format `XX/XX/XXXX` where `X` is a number between 0 to 9, the logic for checking whether this is an acceptable date is contained within the `validAppointmentDate` method.